common/PerfStressShared/PerfStressUtilities.cs

```diff
@@ -49,8 +49,25 @@ internal static Type[] GetOptionTypes(IEnumerable<Type> testTypes)
 
             foreach (var t in testTypes)
             {
-                var baseOptionsType = t.GetConstructors().First().GetParameters()[0].ParameterType;
-                var tb = mb.DefineType(t.Name + "Options", TypeAttributes.Public, baseOptionsType);
+                // Test types must contain a public constructor SampleTest(baseOptionsType, ...)
```

Comment:
This snippet seems to repeat the logic from `PerfProgram`.   You may want to consider breaking the constructor sniffing into its own function and calling it from both places.